### PR TITLE
refactor: Cache moo lexer instances

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Don't create new object each time we need to parse something.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
As far as parsing code is synchronous, there should not be problem

File: lib/manager/bazel/extract.ts [hunk 1]
```diff
@@ -155,6 +155,8 @@ function parseContent(content: string): string[] {
     token = lexer.next();
   }
 
+  lexer.reset();
+
```
Comment (viceice) [inline lib/manager/bazel/extract.ts:159]:
```suggestion
```
I think this is not needed. 🤔

File: lib/manager/cake/index.ts [hunk 2]
```diff
@@ -67,5 +66,6 @@ export function extractPackageFile(content: string): PackageFile {
     }
     token = lexer.next();
   }
+  lexer.reset();
```
Comment (viceice) [inline lib/manager/cake/index.ts:69]:
```suggestion
```

**Review by viceice — COMMENTED** [review]
Yes, as node functions can't be called in parallel (no multithreading in node yet).

The only think happens is the call order depends on async handling, so we need to be sure lexer state is not shared between async functions.

Also some small suggestions.

**Review by viceice — APPROVED** [review]

**Pull request merged**